mypy/newsemanal/semanal.py

```diff
@@ -2270,8 +2270,12 @@ def check_and_set_up_type_alias(self, s: AssignmentStmt) -> bool:
         #     A = float  # OK, but this doesn't define an alias
         #     B = int
         #     B = float  # Error!
-        if existing and (isinstance(existing.node, Var) or
-                         isinstance(existing.node, TypeAlias) and not s.is_alias_def):
+        # Don't create an alias in these cases:
+        if existing and (isinstance(existing.node, Var) or  # existing variable
+                isinstance(existing.node, TypeAlias) and not s.is_alias_def or  # existing alias
+                (isinstance(existing.node, PlaceholderNode) and
+                # TODO: find a more robust way to track the order of definitions.
+                 existing.node.node.line < s.line)):  # or previous incomplete definition
```

Comment:
I believe this to be robust enough in practice, but on equal one would want to also check the column, not?

Reply:
It is more subtle than this. First of all our tracking of columns is bad (for historical reasons); second, using placeholder sources for anything that affects semantics is dangerous because we didn't assign them carefully before.; finally, we need to be sure this works correctly in incremental and daemon modes when files are edited and processed only partially.

Reply:
Is there a plan to fix the columns bug? The daemon thing probably makes it dealbreaker though to go down that path, maybe we should document the currently limited behaviour then, just so that user knows what to expect.